raiden/api/rest.py

```diff
@@ -1028,12 +1033,20 @@ def initiate_payment(
                 target=target_address,
                 amount=amount,
                 identifier=identifier,
+                secret=secret,
+                secret_hash=secret_hash,
             )
-        except (InvalidAmount, InvalidAddress, PaymentConflict, UnknownTokenAddress) as e:
+        except (InvalidAmount, InvalidAddress,
+                PaymentConflict, UnknownTokenAddress) as e:
             return api_error(
                 errors=str(e),
                 status_code=HTTPStatus.CONFLICT,
             )
+        except InvalidSecretOrSecretHash as e:
```

Comment:
Why bad request? For any and all invalid input we use 409 CONFLICT as you can see from the code already.

Reply:
@LefterisJP I think that BAD_REQUEST semantic is better here. Why a wrong input length is a CONFLICT? 
Actually I think that `InvalidAmount`, `InvalidAddress`, and `UnknownTokenAddress` should all result with BAD_REQUEST and not with CONFLICT.
`PaymentConflict` is the only one that justifies `CONFLICT` status.

Please reconsider and let me know your decision. Have no problem to change it.

Reply:
It's a historical thing. If we changed it it would be for all other endpoints too. And I would argue now is not the time to do it. At least not in this PR.

Reply:
Done

Reply:
@LefterisJP just to support what I wrote above - see the below code from `test_restapi.py`
```
def assert_payment_conflict(responses):
    assert all(response is not None for response in responses)
    assert any(
        response.status_code == HTTPStatus.CONFLICT and
        response.json()['errors'] == 'Another payment with the same id is in flight'
        for response in responses
    )
```

Looks like the intention of HTTPStatus.CONFLICT was for the case of `same if in flight`.